### PR TITLE
TextView delegate improved

### PR DESCRIPTION
**Pull request by enchev**

### CODE REVIEW

**Comment by enchev** [comment]
run ci

File: ui/text-view/text-view.ios.ts [hunk 1]
```diff
@@ -6,44 +6,59 @@ import enums = require("ui/enums");
 global.moduleMerge(common, exports);
 
 class UITextViewDelegateImpl extends NSObject implements UITextViewDelegate {
-    public static ObjCProtocols = [UITextViewDelegate];
```
Comment (hshristov) [inline ui/text-view/text-view.ios.ts]: this line is needed.

File: ui/text-view/text-view.ios.ts [hunk 2]
```diff
@@ -6,44 +6,59 @@ import enums = require("ui/enums");
...
+        let impl = <UITextViewDelegateImpl>UITextViewDelegateImpl.new();
+        impl._owner = owner;
+        return impl;
     }
 
     public textViewShouldBeginEditing(textView: UITextView): boolean {
-        this._owner._hideHint();
+        let owner = this._owner.get();
+        if (!owner) {
+            return;
```
Comment (hshristov) [inline ui/text-view/text-view.ios.ts]: Should return true here.

File: ui/text-view/text-view.ios.ts [hunk 3]
```diff
@@ -6,44 +6,59 @@ import enums = require("ui/enums");
...
-        this._owner.dismissSoftInput();
-        this._owner._refreshHintState(this._owner.hint, textView.text);
+        owner.dismissSoftInput();
+        owner._refreshHintState(owner.hint, textView.text);
     }
 
     public textViewDidChange(textView: UITextView) {
+        let owner = this._owner.get();
+        if (!owner) {
+            return;
```
Comment (hshristov) [inline ui/text-view/text-view.ios.ts]: returning here will skip setting 'selectedRange' property. This could possible break UITextView selection.

**Comment by enchev** [comment]
run ci

**Comment by hshristov** [comment]
:+1:

**Pull request merged**